learn/tutorials/adding-typescript-views-to-a-vaadin-14-app/content.adoc

```diff
@@ -0,0 +1,429 @@
+
+.frontend/index.ts
+[source, typescript]
+----
+import {Flow} from '@vaadin/flow-frontend/Flow';
+import {Router} from '@vaadin/router';
+
+import './main-layout'
+
+const {serverSideRoutes} = new Flow({
+  imports: () => import('../target/frontend/generated-flow-imports')
+});
+
+const routes = [
+  {
+    path: '',
+    component: 'main-layout',
+    children: [
+      // fallback to server-side Flow routes if no client-side routes match
+      ...serverSideRoutes
+    ]
+  }
+];
+
+const router = new Router(document.querySelector('#outlet'));
+router.setRoutes(routes);
+----
+
+[WARNING]
+.Move the Java annotations from the main layout to the app shell class
+====
+Because the client-side bootstrapping was introduced in Vaadin 15, the Java main layout does no longer directly correspond to the initial HTML page.
+
+Instead, the `VaadinAppShell` class is introduced for configuring the initial HTML.
+
+From now on, the Java annotations related to the initial HTML should be , such as `@Meta`, `@Viewport`, `@Title`, `@PWA`, should be declared on the `VaadinAppShell` subclass.
+
+.AppShell.java
+[source, java]
+----
+package ...;
+
+import com.vaadin.flow.component.page.VaadinAppShell;
+import com.vaadin.flow.server.PWA;
+import com.vaadin.flow.theme.Theme;
+import com.vaadin.flow.theme.lumo.Lumo;
+
+@Theme(value = Lumo.class)
+@PWA(name = "Bookstore", shortName = "Bookstore")
+public class AppShell extends VaadinAppShell {
+}
+----
+====
+
+Now you can delete the `MainLayout.java` class and all its references.
+
+Rebuild your application and reload the browser to see the changes in effect.
+
+The next step will be purely on the frontend side. Keep your application running, restarting the server for applying the changes is unnecessary. Reloading the browser page is enough.
+
+== Add first TypeScript view
+
+The application structure is now prepared for adding TypeScript views. Let us create one to demonstrate that.
+
+Create `frontend/views/example/example-view.ts` file:
+
+.frontend/views/example/example-view.ts
+[source, typescript]
+----
+import {customElement, html, LitElement, unsafeCSS} from 'lit-element';
+
+import '@vaadin/vaadin-lumo-styles/all-imports';
+
+import {cssFromModule} from "@polymer/polymer/lib/utils/style-gather";
+
+// @ts-ignore
+import styles from './example-view.css';
+
+@customElement('example-view')
+export class MainLayoutElement extends LitElement {
+  static get styles() {
+    return [
+      unsafeCSS(cssFromModule('lumo-typography')),
+      unsafeCSS(styles)
+    ];
+  }
+
+  render() {
+    return html`
+      <h1>Example View</h1>
+      <p>Insert content here...</p>
+    `;
+  }
+}
+----
+
+And the `frontend/views/example/example-view.css` file:
+
+.frontend/views/example/example-view.css
+[source, css]
+----
+:host {
+    display: block;
+    padding: var(--lumo-space-wide-l);
+}
+----
+
+Add a route to the `frontend/index.ts`:
+
+.frontend/index.ts excerpt
+----
```

Comment:
This row should be deleted to render it correctly

Reply:
Done.